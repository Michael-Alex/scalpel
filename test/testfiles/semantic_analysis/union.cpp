union u;

union u
{
	class c
	{
		int i;
	};

	double d;
	int i;
};

