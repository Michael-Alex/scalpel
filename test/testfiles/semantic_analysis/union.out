{
  "id": 0,
  "unions":
  [
    {
      "id": 0,
      "name": "u",
      "classes":
      [
        {
          "id": 0,
          "name": "c",
          "access": "public",
          "destructor":
          {
            "access": "public"
          },
          "variables":
          [
            {
              "id": 0,
              "name": "i",
              "access": "private",
              "type":
              {
                "category": "fundamental type",
                "type": "int"
              }
            }
          ]
        }
      ],
      "destructor":
      {
        "access": "public"
      },
      "variables":
      [
        {
          "id": 1,
          "name": "d",
          "access": "public",
          "type":
          {
            "category": "fundamental type",
            "type": "double"
          }
        },
        {
          "id": 2,
          "name": "i",
          "access": "public",
          "type":
          {
            "category": "fundamental type",
            "type": "int"
          }
        }
      ]
    }
  ],
  "variables":
  [
    {
      "id": 0,
      "name": "u0",
      "type":
      {
        "category": "union",
        "id": 0
      }
    }
  ]
}
